### PR TITLE
ppomppu.co.kr - tracking

### PR DESCRIPTION
**Pull request by piquark6046**
### To avoid invalid pull requests, please check and confirm following terms

- [X] This is not an ad/bug report;
- [X] My code follows the [guidelines](https://github.com/AdguardTeam/AdguardFilters/blob/master/CONTRIBUTING.md) and [syntax](https://kb.adguard.com/general/how-to-create-your-own-ad-filters) of this project;
- [X] I have performed a self-review of my own changes;
- [X] My changes do not break web sites, apps and files structure.

### If the problem does not fall under any category that is listed here, please write a comment below in corresponding section

- [ ] Missed ads or ad leftovers;
- [ ] Website or app doesn't work properly;
- [ ] AdGuard gets detected on a website;
- [X] Missed analytics or tracker;
- [ ] Social media buttons — share, like, tweet, etc;
- [ ] Annoyances — pop-ups, cookie warnings, etc;
- [ ] Filters maintenance.

### Enter the issue address

 - https://www.ppomppu.co.kr/hotdeal/
 - https://ppomppu.co.kr/hotdeal/
 - https://m.ppomppu.co.kr/new/hotdeal/
 - https://www.ppomppu.co.kr/zboard/view.php?id=ppomppu&page=1&divpage=74&no=434667
 - https://m.ppomppu.co.kr/new/bbs_view.php?id=ppomppu&page=1&divpage=74&no=434667&extref=1
 - https://ppomppu.co.kr/zboard/view.php?id=ppomppu&page=1&divpage=74&no=434667

### Add your comment and screenshots

 - https://www.similarweb.com/website/ppomppu.co.kr/#traffic

### Terms

- [X] By submitting this issue, I agree that pull request does not contain private info and all conditions are met

### CODE REVIEW

**Comment by piquark6046** [comment]
Related: https://github.com/AdguardTeam/AdguardFilters/issues/130505

**Comment by Alex-302** [comment]
What exaclty is tracked?

**Comment by piquark6046** [comment]
`https://s.ppomppu.co.kr/?idno=ppomppu_434667&target=aHR0cDovL3d3dy4xMXN0LmNvLmtyL3Byb2R1Y3RzLzQ1Njg1NDg0MDkvc2hhcmU=&encode=on` [^1] exists on https://www.ppomppu.co.kr/zboard/view.php?id=ppomppu&page=1&divpage=74&no=434667.
This encoded URL can be used for tracking visitors.

---

Other tracking URL can be found on https://ppomppu.co.kr/hotdeal/.
<details>
<summary>`li.item.grid-item.23921848` Screenshot</summary>

Screensh232

</details>

`https://www.ppomppu.co.kr/hotdeal/?act=bridge&no=23921848&url=http%3A%2F%2Fclick.linkprice.com%2Fclick.php%3Fm%3Dcoupang%26a%3DA100211316%26l%3D9999%26l_cd1%3DS%26l_cd2%3Dv%26s%3Dmpage%26tu%3Dhttps%253A%252F%252Fwww.coupang.com%252Fvp%252Fproducts%252F7958974%253FitemId%253D11966409657%2526vendorItemId%253D3058658009`
The `url` parameter is used for tracking and advertising. However, These URL encoded with `encodeURIComponent` can be useful for ppomppu users.

`http://click.linkprice.com/click.php?m=coupang&a=A100211316&l=9999&l_cd1=S&l_cd2=v&s=mpage&tu=https%3A%2F%2Fwww.coupang.com%2Fvp%2Fproducts%2F7958974%3FitemId%3D11966409657%26vendorItemId%3D3058658009` (decoded with `decodeURIComponent` once.) [^2][^3]



[^1]: The `target` parameter is encoded with Base64. Can be decoded with `atob`.
[^2]: The homepage of https://linkprice.com says "we operate LinkPrice partnership business and LinkPrice online advertisement business.".
[^3]: The `tu` parameter contains `https://www.coupang.com/vp/products/7958974?itemId=11966409657&vendorItemId=3058658009` (decoded with `decodeURIComponent` twice, in the result).

---

 @Alex-302

**Comment by Alex-302** [comment]
> This encoded URL can be used for tracking visitors.

I see just destination link in that base64 code.

> The `url` parameter is used for tracking and advertising. However, These URL twice-encoded with `encodeURIComponent` can be useful for ppomppu users.

`tu` and `origin` are the parameters of **click.linkprice.com**

**Comment by piquark6046** [comment]
> I see just destination link in that base64 code.
However, The administrator of ppomppu can know how many visitors visit the targeted website.

> `tu` and `origin` are the parameters of click.linkprice.com
Yeap. They are the parameters of click.linkprice.com.
But, The image and information attached on ppomppu.co.kr indicate the URL value of `tu`.

**Comment by piquark6046** [comment]
Ping.

File: SpywareFilter/sections/general_extensions.txt [hunk 1]
```diff
@@ -16,6 +16,8 @@ imdb.com#%#//scriptlet("set-constant", "doWithAds", "noopFunc")
...
 ppomppu.co.kr#%#AG_onLoad(() => { if (/m\.ppomppu\.co\.kr\/new\/bbs_view\.php\?/.test(location.href) || /ppomppu\.co\.kr\/zboard\/view\.php\?/.test(location.href)) { Array.from(document.querySelectorAll("a[href][target='_blank']")).filter(e => /^https:\/\/s\.ppomppu\.co\.kr(\/|)\?idno=/.test(e.href)).forEach((o) => { o.href = atob(new URLSearchParams(o.href).get("target").replace("\\", ""))})}});
+! https://github.com/AdguardTeam/AdguardFilters/issues/132024
+revv.co.in#%#(()=>{window.google_optimize={get(){}}})();
```
Comment (Alex-302) [inline SpywareFilter/sections/general_extensions.txt:20]: Wong commit?

**Comment by Alex-302** [comment]
> However, The administrator of ppomppu can know how many visitors visit the targeted website directly by logging number of requests about the page.

I think it is too paranoid theory)

**Comment by piquark6046** [comment]
> I think it is too paranoid theory)

Okay. however, I want to modify the `href` of them to prevent their data collection.
Anyway, they will not have to provide the modified address if they don't have a related intention.

If you really want to discard this changes, please close this PR.

**Review by zloyden — APPROVED** [review]

File: SpywareFilter/sections/general_extensions.txt [hunk 2]
```diff
@@ -14,7 +14,7 @@ imdb.com#%#//scriptlet("set-constant", "doWithAds", "noopFunc")
...
+ppomppu.co.kr#%#AG_onLoad(() => { if (/^https:\/\/(www\.|m\.|)ppomppu\.co\.kr\/(new\/|)hotdeal\//g.test(document.location.href) && document.querySelector(".hotDeal_goods > ul#hotdeal") != null) { var k = (n) => { if(/\/\/click\.linkprice\.com\/click\.php\?/.test(n)) { return new URLSearchParams(n).get("tu") } else { return n } }; new MutationObserver((o) => { Array.from(o[0].addedNodes).filter(e => /[0-9]{7,9}$/.test(e.className) && /ppomppu\.co\.kr\/(new\/|)hotdeal\/\?/.test(e.querySelector(".view > a").href)).forEach((e) => { e.querySelector(".view > a").href = e.querySelector("p > a[href]:not([class])").href = k(new URLSearchParams(e.querySelector(".view > a").href).get("url"))})}).observe(document.querySelector(".hotDeal_goods > ul#hotdeal"), { childList: true })}});
```
Comment (zloyden) [inline SpywareFilter/sections/general_extensions.txt:17]: > && document.querySelector(".hotDeal_goods > ul#hotdeal") != null

I guess we can use it without checking `null`, just

```
&& document.querySelector(".hotDeal_goods > ul#hotdeal")
```

Comment (piquark6046) [reply SpywareFilter/sections/general_extensions.txt]: Done.

**Review by AdamWr — APPROVED** [review]

**Review by zloyden — APPROVED** [review]

**Pull request merged**